elements/howto-menu/howto-menu.js

```diff
@@ -0,0 +1,115 @@
+/*
+ * A menu is a widget that offers a list of choices to the user,
+ * such as a set of actions or functions. A menu is usually opened,
+ * or made visible, by activating a menu button, choosing an item in a menu
+ * that opens a sub menu, or by invoking a command, such as Shift + F10 in
+ * Windows, that opens a context specific menu.
+ *
+ * The element that opens the menu is referenced with aria-labelledby.
+ * TODO(devnook): Should it be aria-labelledby or aria-describedby?
+ *
+ * See: https://www.w3.org/TR/wai-aria-practices-1.1/#menu
+ */
+
+ class HowtoMenu extends HTMLElement {
+
+   /**
+    * The constructor does work that needs to be executed _exactly_ once.
+    */
+   constructor() {
+     super();
+
+     this.KEY_CODES_ = {
+       ARROW_DOWN: 40,
+       ARROW_UP: 38,
+       ESCAPE: 27,
+     };
+     this.children_ = null;
+     // TODO: Consider better naming?
+     this.parent_ = document.getElementById(
+         this.getAttribute('aria-labelledby'));
+   }
+
+   moveFocus_(fromEl, toEl) {
+     fromEl.setAttribute('tabindex', -1);
+     toEl.setAttribute('tabindex', 0);
+     toEl.focus();
+   }
+
+   handleBlur_(e) {
+     e.target.setAttribute('tabindex', -1);
```

Comment:
I am not sure if `blur` bubbles, but if someone does crazy stuff like putting an `<input>` in their menu, this might be do unexpected things. I think a check if `e.target` is actually a `<howto-menuitem>` should suffice.